satpy/readers/fci_l1c_nc.py

```diff
@@ -687,16 +702,9 @@ def calibrate_rad_to_refl(self, radiance, key):
                 "cannot produce reflectance for {:s}.".format(measured))
             return radiance * np.float32(np.nan)
 
-        sun_earth_distance = np.mean(
+        sun_earth_distance = np.nanmean(
             self.get_and_cache_npxr("state/celestial/earth_sun_distance")) / 149597870.7  # [AU]
```

Comment:
I have some issue with the `np.nanmean(self._get_aux_data_lut_vector("earth_sun_distance"))` I get this value `0.2005376136680534` which is bellow 0.9 U.A , right now the `TestFCIL1cNCReaderBadDataFromIDPF.test_handling_bad_earthsun_distance`  is failing because i have removed this line 

```python 
# TODO remove this check when old versions of IDPF test data (<v5) are deprecated.
        if sun_earth_distance < 0.9 or sun_earth_distance > 1.1:
            logger.info("The variable state/celestial/earth_sun_distance contains unexpected values"
                        "(mean value is {} AU). Defaulting to 1 AU for reflectance calculation."
                        "".format(sun_earth_distance))
            sun_earth_distance = 1
```

Should i change the test ?

Reply:
could you please check why we get that low number even with the nanmean? Which data is producing that?
In principle, we should not need the `if` checking the quality of the `sun_earth_distance` anymore, so you can also get rid of the tests.

Reply:
It is determined by the mock data here 
```python 
class FakeFCIFileHandlerWithBadIDPFData(FakeFCIFileHandlerFDHSI):
    """Mock bad data for IDPF TO-DO's."""

    def _get_test_content_all_channels(self):
        data = super()._get_test_content_all_channels()
        data["data/vis_06/measured/x"].attrs["scale_factor"] *= -1
        data["data/vis_06/measured/x"].attrs["scale_factor"] = \
            np.float32(data["data/vis_06/measured/x"].attrs["scale_factor"])
        data["data/vis_06/measured/x"].attrs["add_offset"] = \
            np.float32(data["data/vis_06/measured/x"].attrs["add_offset"])
        data["data/vis_06/measured/y"].attrs["scale_factor"] = \
            np.float32(data["data/vis_06/measured/y"].attrs["scale_factor"])
        data["data/vis_06/measured/y"].attrs["add_offset"] = \
            np.float32(data["data/vis_06/measured/y"].attrs["add_offset"])

        data["state/celestial/earth_sun_distance"] = xr.DataArray(da.repeat(da.array([30000000]), 6000))
        return data
```

Reply:
ah ok, that's expected then, you can get rid of the mockers for bad data and the according tests then